libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1367,6 +1368,30 @@ bool RC_Channel::do_aux_function(const aux_func_t ch_option, const AuxSwitchPos
            }
         break;
     }
+    
+    case AUX_FUNC::MAG_CAL: {
+        Compass &compass = AP::compass();
+        switch (ch_flag) {
+        case AuxSwitchPos::LOW:
+            compass.cancel_calibration_all();
+            break;
+        case AuxSwitchPos::MIDDLE:
+            // nothing
+            break;
+        case AuxSwitchPos::HIGH:
+            if (!hal.util->get_soft_armed()) {
+                const bool retry = true;
+                const bool autosave = true;
+                const float delay = 5.0;
```

Comment:
what does this delay do? at start? or on retry?

Reply:
just copying what the old stick gesture startup used....I think it was to give a person time to pick up the vehicle from the ground or other objects before starting the cal...not sure its critical